### ISSUE TITLE
Consider backend for passwordstore?

### ISSUE THREAD

**Issue by javierluraschi** [r-lib/keyring#46]
I think it would be worth considering https://www.passwordstore.org/ for Linux services.

`backend = "env"` is really easy to set up but not ideal in some situations, `backend = "vault"` would be harder to set up but very secure. A `backend = "linux-pass"` could sit between those two for users that want additional security, but perhaps not a full-blown solution.

**Comment by gaborcsardi** [comment]
I like this, except that it seems that there is no library for it, so we'll need to call out to the command line tool. But it can work.

The only thing we need to work out is to store the master password in the session somehow. Probably we should just use `gpg-agent` as they suggest.

**Comment by javierluraschi** [comment]
From @jmcphers: `pass` is a script that makes it more convenient to store passwords in GPG files and retrieve them somewhat interactively. For this use case, and presuming we want to depend on GPG for encryption, I think we'd be better off just depending on GPG directly as pass's syntax sugar is geared almost entirely towards end users.

**Comment by gaborcsardi** [comment]
Makes sense. Although is this case we don't really need GPG, either, just openssl or libsodium, both available as an R packages.

I wonder if we want a generic way of caching master keys in the session.

**Comment by jmcphers** [comment]
The nice thing about gpg is that it lets you add secrets (a) without the secret key (since it uses asymmetric encryption), and (b) without R (since it has a nice command-line interface, too). I think this might be useful for cases like Connect where you won't necessarily be using the keyring package to write the secret, just to read it. I think you can do the same with openssl, though I'm less familiar with it.

**Comment by gaborcsardi** [comment]
Asymmetric crypto is not needed/used for the keychain. E.g. you cannot put items on the keychain without unlocking it. 

GPG is OK well, although we would probably not use as symmetric crypto. openssl and sodium both have bindings to all languages and they are much easier to program than gpg.
